src/services/formatting/formatting.ts

```diff
@@ -432,13 +432,12 @@ namespace ts.formatting {
             if (leadingTrivia) {
                 indentTriviaItems(leadingTrivia, indentation, /*indentNextTokenOrTrivia*/ false,
                     item => processRange(item, sourceFile.getLineAndCharacterOfPosition(item.pos), enclosingNode, enclosingNode, /*dynamicIndentation*/ undefined!));
+                if (options.trimTrailingWhitespace !== false) {
+                    trimTrailingWhitespacesForRemainingRange(leadingTrivia);

```

Comment:
why don't we need to call `trimTrailingWhitespacesForRemainingRange` in the above case when `isOnToken()` is true?

Reply:
`trimTrailingWhitespacesForRemainingRange()` should only be use for leading trivia I believe. By the time `isOnToken()` is false that means we're at the end of the file or at trivia after processing the previous nodes or outside the original range; otherwise we'll require to process the node like we're doing in on the previous lines.

Hope this clarifies it.